### PR TITLE
Add order cancellation

### PR DESCRIPTION
**Pull request by PiotrZaranek**

### CODE REVIEW

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs
Comment (dawiddworak88) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs:619]: **question:** Isn't there a check missing that will send email only when the status has been set to cancelled? I'm concerned that it would be triggered for any status change

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs
Comment (PiotrZaranek) [reply be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs:619]: I added email notifications when `serviceModel.OrderStatusId == OrderStatusesConstants.CanceledId`. You can check line 595 for the condition.

Comment (MarcinKalinowski1) [reply be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs:619]: if it is a change from the client's side, it is not necessary because the client can only set the status to 'Cancelled', but if it is a change from the admin's side, it seems that there should be a check here because he can change the status to something other than 'Cancelled'

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.de.resx
Comment (SebaKas21) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.de.resx:300]: **question:** Is this property used somewhere? because I'm afraid it is and we'll have to modify the appropriate template

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.pl.resx
Comment (SebaKas21) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.pl.resx]: **suggestion:** Missing space between words

File: be/src/Project/Web/Buyer/Buyer.Web/Areas/Orders/ApiControllers/OrderStatusApiController.cs
Comment (SebaKas21) [inline be/src/Project/Web/Buyer/Buyer.Web/Areas/Orders/ApiControllers/OrderStatusApiController.cs]: **suggestion:** Redundant usings

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs
Comment (SebaKas21) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs]: **suggestion (non-blocking):** you can use linkGenerator

File: fe/projects/Seller.Portal/src/areas/Orders/components/OrderForm/OrderForm.js
Comment (SebaKas21) [inline fe/projects/Seller.Portal/src/areas/Orders/components/OrderForm/OrderForm.js]: **question:** Why you modify this suggestion?

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.de.resx
Comment (PiotrZaranek) [reply be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.de.resx:300]: I only changed the names in the resource files; the template variables remain the same as before these changes.

File: fe/projects/Seller.Portal/src/areas/Orders/components/OrderForm/OrderForm.js
Comment (PiotrZaranek) [reply fe/projects/Seller.Portal/src/areas/Orders/components/OrderForm/OrderForm.js]: Because from be property name has value like this "Anton (An_01)".

image

Line 52.

**Comment by SebaKas21** [comment]
[like]  Sebastian Kasperek reacted to your message:
________________________________

**Comment by SebaKas21** [comment]
[like]  Sebastian Kasperek reacted to your message:
________________________________

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs
Comment (PiotrZaranek) [reply be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs]: I cannot use LinkGenerator because Ordering.Api is separate from Buyer.Web, and I get a null result. However, I moved the URLs to the constants file.

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs
Comment (dawiddworak88) [reply be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs:619]: Thanks @MarcinKalinowski1, we'll improve it in the future, let's leave it for now

File: fe/projects/Seller.Portal/src/areas/Orders/components/OrderForm/OrderForm.js
Comment (dawiddworak88) [reply fe/projects/Seller.Portal/src/areas/Orders/components/OrderForm/OrderForm.js]: we should not modify it, please revert

File: fe/projects/AspNetCore/src/areas/Orders/components/OrderItemForm/OrderItemForm.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Orders/components/OrderItemForm/OrderItemForm.js:22]: why to touch it, is it connected in any way with product cancellation?

File: be/src/Project/Web/Buyer/Buyer.Web/Areas/Orders/ApiControllers/OrderStatusApiController.cs
Comment (dawiddworak88) [inline be/src/Project/Web/Buyer/Buyer.Web/Areas/Orders/ApiControllers/OrderStatusApiController.cs]: typo, it is cancelled with double 'l', please check everywhere in the PR

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: fe/projects/AspNetCore/src/areas/Orders/components/OrderItemForm/OrderItemForm.js
Comment (PiotrZaranek) [reply fe/projects/AspNetCore/src/areas/Orders/components/OrderItemForm/OrderItemForm.js:22]: Yes, I added a check for the order item status in the API controller, so I needed to retrieve the current status of the order item.

**Review by dawiddworak88 — APPROVED** [review]
🎖️ Thank you @PiotrZaranek Let's merge it and ask for retesting after merged deployment to test

**Comment by dawiddworak88** [comment]
@SebaKas21, would you have a look and approve if everything is alright?

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx
Comment (SebaKas21) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx]: **suggestion:** Should be cancelled

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx
Comment (SebaKas21) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx]: **suggestion:** Should be cancelled

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx
Comment (SebaKas21) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx]: **suggestion:** Should be cancelled

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx
Comment (SebaKas21) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/OrderResources.en.resx]: **suggestion:** Should be cancelled

**Review by SebaKas21 — CHANGES_REQUESTED** [review]
Hi @PiotrZaranek 

Fix typo and everything will be okay

**Review by SebaKas21 — APPROVED** [review]

**Pull request merged**
